sub/sd_ass.c

```diff
@@ -288,46 +288,53 @@ static int init(struct sd *sd)
     return 0;
 }
 
+static inline bool is_escaped(const char *str, size_t pos)
+{
+    size_t count = 0;
+    while (pos > 0 && str[--pos] == '\\')
+        ++count;
+    return count % 2;
+}
+
 // Check if subtitle has events that would cause it to be animated inside {}
-static bool is_animated(char *s)
+static bool is_animated(const char *str)
 {
-    bool in_tag = false;
-    bool valid_event = false;
-    bool valid_tag = false;
-    while (*s) {
-        if (!in_tag && s[0] == '{')
-            in_tag = true;
-        if (s[0] == '\\') {
-            s++;
-            if (!s[0])
+    const char *begin = str;
+    while ((begin = strchr(begin, '{'))) {
+        if (!begin)
+            return false;
+
+        if (is_escaped(str, begin - str)) {
+            ++begin;
+            continue;
+        }
+
+        const char *end = begin;
+        while ((end = strchr(end, '}'))) {
```

Comment:
No need for this loop; you can just do `const char *end = strchr(begin, '}');` (in fact this is exactly what libass does), and then just `if (!end) return false;` (as you have now).